#!/usr/bin/env python3.8
PACKAGE = "imitation_learning_ros_package"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# gen.add("rate_fps",    int_t,  0, "Update rate of controller", 5,  1, 100)
# gen.add("max_input", double_t, 0, "Maximum absolute control value",    1, 0.01,   10)

# parameter set simulation:
# Px=1.0, Ix=0.3, Dx=0.3, Py=1.0, Iy=0.3, Dy=0.3, Pz=1.0, Iz=0.3, Dz=0., Theta=1.0
# parameter set real:
# Px=0.1, Ix=0., Dx=0.1, Py=0.1, Iy=0., Dy=0.1, Pz=1.0, Iz=0.1, Dz=0., Theta=1.0

gen.add("Kp_x", double_t, 0, "PID",    0.1, 0.0,   10)
gen.add("Ki_x", double_t, 0, "PID",    0., 0.0,   10)
gen.add("Kd_x", double_t, 0, "PID",    0.1, 0.0,   10)
gen.add("Kp_y", double_t, 0, "PID",    0.1, 0.0,   10)
gen.add("Ki_y", double_t, 0, "PID",    0., 0.0,   10)
gen.add("Kd_y", double_t, 0, "PID",    0.1, 0.0,   10)
gen.add("Kp_z", double_t, 0, "PID",    1., 0.0,   10)
gen.add("Ki_z", double_t, 0, "PID",    0., 0.0,   10)
gen.add("Kd_z", double_t, 0, "PID",    0., 0.0,   10)
gen.add("K_theta", double_t, 0, "PID", 1.0, 0.0,   10)


exit(gen.generate(PACKAGE, "imitation_learning_ros_package", "pid"))
